### PR TITLE
Add Pipes

### PR DESCRIPTION
**Pull request by drg101**
Added pipe.js and documentation/tests for it.

### CODE REVIEW

**Review by Marc-9 — APPROVED** [review]
Looks good to me, I would like to see some example code of how this would look in action

File: src/dependencies/pipe.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            return true;
+        }
+        return false;
+    },
+
+    //private functions --------------
+    /**
+      * Finds the index of a pipe
+      * @memberof Pipe
+      * @method pipe
```
Comment (pierce-smith1) [inline src/dependencies/pipe.js:68]: The method name in the docs here is wrong.
Other than that I don't see a problem. Tests pass, code is clean.

**Comment by drg101** [comment]
Made the flyTo animation just go to the center of where the user clicks. This fixes the bug for the icons (on my end), whilst removing the zooming which I think was what was annoying about the old flyTo.

**Comment by drg101** [comment]
I think this allows the map to feel as "responsive" as possible whilst also being super simple.

**Pull request merged**